### PR TITLE
Returning Correct Orbit of Moon (GCRS)

### PR DESCRIPTION
**Pull request by shreyasbapat**
`358331 x 414655 km x 20.8 deg orbit around Earth (♁)`
I have some doubts before this. Where should we include ``solar_system_ephemeris.set("jpl")``? Is it safe to keep in poliastro? Or the user must use it on their own?
Solves #382

### CODE REVIEW

**Comment by shreyasbapat** [comment]
If not, I feel we must include this somewhere in the docs. I more thing that is missing to be included in docs is the plotly jupyterlab extension.

File: src/poliastro/twobody/orbit.py [hunk 1]
```diff
@@ -162,7 +168,19 @@ def from_body_ephem(cls, body, epoch=None):
             warn("Input time was converted to scale='tdb' with value "
                  "{}. Use Time(..., scale='tdb') instead."
                  .format(epoch.tdb.value), TimeScaleWarning)
+        if body == Moon:
+            solar_system_ephemeris.set("jpl")
```
Comment (Juanlu001) [inline src/poliastro/twobody/orbit.py]: We should don't do this: it affects the whole program without noticing, as you saw by the test failures.

File: src/poliastro/twobody/orbit.py [hunk 2]
```diff
@@ -162,7 +168,19 @@ def from_body_ephem(cls, body, epoch=None):
...
         r, v = get_body_barycentric_posvel(body.name, epoch)
+        if body == Moon:
+            moon_icrs = ICRS(x=r.x, y=r.y, z=r.z, v_x=v.x, v_y=v.y, v_z=v.z,
+                             representation=CartesianRepresentation, differential_type=CartesianDifferential
+                             )
+            moon_gcrs = moon_icrs.transform_to(GCRS(obstime=epoch))
+            moon_gcrs.representation = CartesianRepresentation
+            return cls.from_vectors(
```
Comment (Juanlu001) [inline src/poliastro/twobody/orbit.py]: You can avoid having two `return` statements by carefully setting the attractor, the position and velocity inside this `if` branch.

File: src/poliastro/twobody/orbit.py (hunk 1 above)
Comment (shreyasbapat) [reply src/poliastro/twobody/orbit.py]: Yeah Sure!

File: src/poliastro/twobody/orbit.py (hunk 2 above)
Comment (shreyasbapat) [reply src/poliastro/twobody/orbit.py]: Yes, I did that! Thanks for pointing this out!

**Comment by shreyasbapat** [comment]
That's why, I saw multiple tests failing xD. Sorry for the trouble

**Comment by shreyasbapat** [comment]
Is it still me xD?

**Comment by Juanlu001** [comment]
Yes, but in this case the tests need to be update, since they rely on the old behavior I'd say.

**Comment by Juanlu001** [comment]
@nikita-astronaut thoughts?

**Comment by shreyasbapat** [comment]
I didn't actually understand what this means:

> the tests need to be update, since they rely on the old behavior

Can you elaborate a bit?

**Comment by Juanlu001** [comment]
At least the orbit of the Moon makes sense now!

```
In [5]: solar_system_ephemeris.set("jpl")
Out[5]: <ScienceState solar_system_ephemeris: 'jpl'>

In [6]: Orbit.from_body_ephem(Moon)
Out[6]: 360195 x 409425 km x 20.8 deg orbit around Earth (♁)
```

**Comment by Juanlu001** [comment]
The problem is here:

https://github.com/poliastro/poliastro/blob/da6bd151588eac259fc28dd2380c5ed467e6f915/src/poliastro/ephem.py#L10-L19

You have to make another special case there for the Moon, I think.

**Comment by Juanlu001** [comment]
Also, this will need a rebase on current master.

**Comment by shreyasbapat** [comment]
Something of this kind?
screenshot from 2018-07-20 01-37-23

**Comment by Juanlu001** [comment]
Circle CI is having an outage https://status.circleci.com/ Will look at this again in some hours

**Comment by Juanlu001** [comment]
The coverage increased even without adding a new test, but that's because the perturbations tests are already covering this. Thanks @shreyasbapat and merging!

**Pull request merged**